app/decorators/user_decorator.rb

```diff
@@ -116,6 +116,16 @@ def devices?
     !recent_devices.empty?
   end
 
+  def last_signed_in_at
+    user.devices.order(last_used_at: :desc).first&.last_used_at
+  end
+
+  def second_last_signed_in_at
+    user.events.where(
+      event_type: 'sign_in_after_2fa'
+    ).order(id: :desc).pluck(:created_at).second
```

Comment:
```suggestion
    ).order(created_at: :desc).pluck(:created_at).second
```

I think `created_at` is preferable for the sort, and we already have an index on (user_id, created_at) as a bonus.

Reply:
During testing, I think I had some specific reason for preferring id. This has been changed.